File: fastify-ts-bun-backend/src/routes/ping.route.ts
import type { FastifyInstance } from 'fastify';

export async function registerPingRoutes(app: FastifyInstance): Promise<void> {
  app.get('/api/v1/ping', async () => ({
    success: true,
    data: {
      message: 'pong',
      version: 'v1',
    },
  }));
}

File: fastapi-python-backend/src/routes/ping_route.py
from fastapi import APIRouter

ping_router = APIRouter(prefix='/api/v1', tags=['ping'])


@ping_router.get('/ping')
async def ping() -> dict[str, object]:
    return {
        'success': True,
        'data': {
            'message': 'pong',
            'version': 'v1',
        },
    }
